### ISSUE TITLE
Collections.Set To Dictionary not possible to use numeric keys like with BuiltIn.Create Dictionary

### ISSUE THREAD

**Issue by HelioGuilherme66** [robotframework/robotframework#2320]
All is OK with `&{Dict}=    BuiltIn.Create Dictionary    ${1}=one    one=uno    uno=um    um=ein    ein=un`
But fails with `Collections.Set To Dictionary    ${Dict}    two=dois    ${2}=two`
with: 	FAIL 	Argument names must be strings.

**Comment by pekkaklarck** [comment]
This is certainly true. It's not really a bug because it is also the documented behavior of the keyword. The general way to add set dictionary items is using

    Set To Dictionary    ${dict}    key    value    ${2}    two

This is obviously highly inconsistent with how `Create Dictionary` works. It currently still supports

    ${dict} =    Create Dictionary    key    value

but that approach is deprecated in favor of

    ${dict} =    Create Dictionary    key=value

There are various things we could do to remove this inconsistency:

1. Support non-string keys with `Set To Dictionary`. Notice that the underlying reason they don't work is that `**kwargs` syntax in Python supports only string keys. Avoiding that limitation requires changing how arguments are resolved in general. Currently only some keywords in BuiltIn use that approach and there is no clean API to do that. Cleaning up this stuff is the topic of #2190. Required changes may be too big to be done in RF 3.0.x versions.

2. Remove deprecation from `Create Dictionary` to pass keys and values as separate items. Then there would at least be a common way to create and modify dicts. Not forcing users to switch dictionary creating to the `key=value` format would probably be a good idea in general. This could be done already in RF 3.0.1 but would require a separate issue.

**Comment by pekkaklarck** [comment]
Decisions:

1. Try to make `Set To Dictionary` work with all valid dictionary keys similarly as `Create Dictionary`. This requires too big changes to be done in RF 3.0.x but hopefully can be done in RF 3.1.

2. Remove deprecation from passing keys and values individually for `Create Dictionary`. I submitted #2386 about that and targeted it for RF 3.0.1.

**Comment by pekkaklarck** [comment]
RF 3.1 is so long overdue and this is so low priority issue that I move this to RF 3.2 scope. After #2386 included in RF 3.0.1 there's also a non-deprecated uniform way to use `Crate Dictionary` and `Set To Dictionary`.
